### DISCUSSION TITLE
q-toggle: emit the new value when the animation ends

### DISCUSSION THREAD

**Discussion by damvaz** [quasarframework/quasar#16437] [General - Components / Directives / etc]
### What happened?

In componente QToggle, a `@update-model-value` event handler can perform a very heavy task (like updating a Pinia store or something similar job).

When that happens, there is a delay in the component's animation.

Only when the heavy task is finished, the animation continues.

### What did you expect to happen?

The toggle animation should end. 

Then the value change event should just be emitted in the component.

### Reproduction URL

https://codepen.io/damianvazquezJS/pen/LYMXGzZ?editors=0010

### How to reproduce?

1. Go to the provided reproduction link
2. Click on the toggle
3. Watch the delay in the animation/transition of thumb element

### Flavour

Quasar CLI with Vite (@quasar/cli | @quasar/app-vite)

### Areas

Components (quasar)

### Platforms/Browsers

Firefox, Chrome

**Comment by metalsadman** [comment]
you are blocking the main thread

**Reply by damvaz** [reply]
Thanks for your response @metalsadman!

The main thread is blocking, but not from the component implementation side. 

This is a strange case of a side effect of listening to the change event.  Although it is true that it could be solved with some tricks, such as workers, that is not always possible.  

Therefore, IMHO, internally it is the component that should solve this problem.  for example, **waiting for the animation to finish and then emitting the change event.**

Otherwise, its implementation becomes very "dirty". 😔

**Comment by damvaz** [comment]
The same behavior is expected for other components that emit the event `update:modelValue`, for example, `q-checkbox`.
